core/src/main/scala/org/apache/spark/metrics/ExecutorMetricType.scala

```diff
@@ -137,7 +138,9 @@ case object GarbageCollectionMetrics extends ExecutorMetricType with Logging {
 
   override private[spark] def getMetricValues(memoryManager: MemoryManager): Array[Long] = {
     val gcMetrics = new Array[Long](names.length) // minorCount, minorTime, majorCount, majorTime
-    ManagementFactory.getGarbageCollectorMXBeans.asScala.foreach { mxBean =>
+    val mxBeans = ManagementFactory.getGarbageCollectorMXBeans.asScala
+    gcMetrics(4) = mxBeans.map(_.getCollectionTime).sum
```

Comment:
Is `gcMetrics(4)` strictly equal to `gcMetrics(1) + gcMetrics(3)`?

Reply:
in common case they are the same, but when we use nonBuiltInCollectors,
"gcMetrics(1) + gcMetrics(3)"  is zero, but the actual total gc time  is not zero

Reply:
Why not simply rely on MinorGCTime + MajorGCTime instead ?
Custom collectors are not handled in general currently in spark anyway - is there a reason to include them as well ?

Reply:
we should consistently apply the computation of totalGcTime both in driver side and  executor side

Reply:
since the totalGCTime of the real executor apply the sum of all collectionTime

Reply:
Sonds good.